base_external_api/models/external_api_config.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2025 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import datetime
+
+import requests
+
+from odoo import api, fields, models
+from odoo.modules.registry import Registry
+
+from odoo.addons.queue_job.exception import RetryableJobError
+
+
+class ExternalApiConfig(models.Model):
+    _name = "external.api.config"
+    _description = "Defines a external api used by the system and its connection data"
+
+    name = fields.Char(required=True)
+
+    base_url = fields.Char(
+        string="Base URL",
+        required=True,
+        help="Common URL of all the endpoints of the external API. "
+        "Must end with a backslash.",
+    )
+    state = fields.Selection(
+        selection=[
+            ("test", "Test"),
+            ("production", "Production"),
+            ("disabled", "Disabled"),
+        ],
+        default="test",
+        required=True,
+        help="Only production connections will be executed",
+    )
+    enable_logs = fields.Boolean(
+        default=True,
+    )
+    protocol = fields.Selection(
+        selection=[("http", "HTTP / HTTPS")], default="http", required=True
+    )
+    authorization = fields.Selection(
+        string="Auth method",
+        selection=[("none", "None"), ("basic", "Basic Auth"), ("apikey", "API Key")],
+        default="none",
+        required=True,
+    )
+
+    auth_basic_user = fields.Char(string="User")
+    auth_basic_passwd = fields.Char(string="Password")
+
+    auth_apikey_key = fields.Char(string="Header", default="api-key")
+    auth_apikey_value = fields.Char(string="APIKey value")
+
+    @api.model
+    def _parse_url(self, url):
+        url = url.strip()
+        if not url[-1] == "/":
+            url += "/"
+        if url[0] == "/":
+            url = url[1:]
+        return url
+
+    @api.onchange("base_url")
```

Comment:
why onchange and not depends?

Reply:
I think you a field can not depend on himself

Reply:
Changed